### PR TITLE
Fix phpstan errors for tests

### PR DESCRIPTION
**Pull request by VincentLanglet**
.

### CODE REVIEW

**Comment by VincentLanglet** [comment]
@phansys I get an error with
```
        if (!$this->httpMethodParameterOverride && Request::getHttpMethodParameterOverride()) {
            $disableHttpMethodParameterOverride = \Closure::bind(static function (): void {
                self::$httpMethodParameterOverride = false;
            }, null, Request::class);
            $disableHttpMethodParameterOverride();
        }
```
Static access to instance property Sonata\AdminBundle\Tests\Controller\CRUDControllerTest::$httpMethodParameterOverride.

I don't really understand this code ; can you confirm it's valid ?

**Comment by VincentLanglet** [comment]
I need https://github.com/psalm/psalm-plugin-symfony/pull/160
and I need to solve https://github.com/phpstan/phpstan-phpunit/issues/87

**Comment by phansys** [comment]
>I don't really understand this code ; can you confirm it's valid ?

I think yes. This is an example about how it works: https://3v4l.org/LcsM6.

**Comment by VincentLanglet** [comment]
> > I don't really understand this code ; can you confirm it's valid ?
> 
> I think yes. This is an example about how it works: https://3v4l.org/LcsM6.

Ok, I created an issue: https://github.com/phpstan/phpstan/issues/4734

**Comment by VincentLanglet** [comment]
The lint issue is related to https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5505

**Comment by VincentLanglet** [comment]
Now I just need https://github.com/sonata-project/dev-kit/pull/1473

File: phpstan.neon.dist [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
-        - src/Resources/**.tpl.php
+        - tests
+    excludePaths:
+        analyseAndScan:
+            # temporarily ignore template files
+            - src/Resources/**.tpl.php
+            # BC-layer is not supported by phpstan https://github.com/phpstan/phpstan/issues/3542
+            - tests/Fixtures/StubTranslator.php
+    ignoreErrors:
+        - '#Trying to mock an undefined method [a-zA-Z]*\(\) on class stdClass\.#'
```
Comment (jordisala1991) [inline phpstan.neon.dist:15]: Maybe we should add the phpstan baseline, just like with psalm?

Comment (VincentLanglet) [reply phpstan.neon.dist]: Done

**Review by jordisala1991 — APPROVED** [review]
Nice work! It would be good to have a changelog for the things changed on the src folder

**Comment by VincentLanglet** [comment]
> Nice work! It would be good to have a changelog for the things changed on the src folder

I think it's only phpdoc

**Review by phansys — APPROVED** [review]

**Pull request merged**